lib/qbo_api/connection/oauth2.rb

```diff
@@ -10,15 +54,47 @@ def self.included(*)
 
     def default_attributes
       super.merge!(
-        access_token: nil
+        access_token:   nil,
+        refresh_token:  nil,
+        client_id:      defined?(CLIENT_ID) ? CLIENT_ID : nil,
+        client_secret:  defined?(CLIENT_SECRET) ? CLIENT_SECRET : nil,
+        refresh_proc:   nil
       )
     end
+
     def add_oauth2_authorization_middleware(conn)
       conn.request :oauth2, access_token, token_type: 'bearer'
+      conn.use FaradayMiddleware::OAuth2Refresh, self if refresh_token
```

Comment:
if there's no refresh token, then it's never added?

Reply:
Yes, that's the idea. If you're just initializing with an `access_token` and no `refresh_token`, the refresh middleware is not added. Do you think it should be added regardless?

Reply:
Yeah. I can't imagine why you wouldn't want to handle it.  I suppose it could be considered a breaking change, so maybe the 'refresh middleware' should be opt in for this version